pkg/operation/etcdencryption/encryptionconfiguration.go

```diff
@@ -0,0 +1,218 @@
+	}
+	var aesConfig *apiserverconfigv1.AESConfiguration
+	if ec.Resources[0].Providers[0].AESCBC != nil {
+		aesConfig = ec.Resources[0].Providers[0].AESCBC
+	} else {
+		aesConfig = ec.Resources[0].Providers[1].AESCBC
+	}
+	ecEncryptionProviderAESCBCKeyLenA := len(aesConfig.Keys)
+	if ecEncryptionProviderAESCBCKeyLenA < ecEncryptionProviderAESCBCMinKeyLenE {
+		return false, fmt.Errorf("unexpected number of keys in encryption provider aescbc of the EncryptionConfiguration found. Expected are at least %v key(s)", ecEncryptionProviderAESCBCMinKeyLenE)
+	}
+	keyNameMap := make(map[string]bool)
+	for _, key := range aesConfig.Keys {
+		ok, err := isKeyConsistent(key)
+		if (err != nil) || (!ok) {
+			return false, fmt.Errorf("inconsistent key in encryption provider aescbc of the EncryptionConfiguration found")
+		}
+		_, ok = keyNameMap[key.Name]
+		if ok {
+			return false, fmt.Errorf("two or more keys (%v) with same timestamp found in encryption provider aescbc of the EncryptionConfiguration", key.Name)
+		}
+		keyNameMap[key.Name] = true
+	}
+	return true, nil
+}
+
+// Equals checks whether the provided encryption configurations are equal.
+func Equals(ec1 *apiserverconfigv1.EncryptionConfiguration, ec2 *apiserverconfigv1.EncryptionConfiguration) (bool, error) {
+	ec1YamlBytes, err := ToYAML(ec1)
+	if err != nil {
+		return false, fmt.Errorf("error when converting EncryptionConfiguration to yaml: %v", err)
+	}
+	ec1YamlString := string(ec1YamlBytes)
+	ec2YamlBytes, err := ToYAML(ec2)
+	if err != nil {
+		return false, fmt.Errorf("error when converting EncryptionConfiguration to yaml: %v", err)
+	}
+	ec2YamlString := string(ec2YamlBytes)
+	return ec1YamlString == ec2YamlString, nil
+}
+
+// IsActive checks whether the EncryptionConfiguration is active, i.e. whether the provider
+// identity is NOT the first in the list of providers.
+func IsActive(ec *apiserverconfigv1.EncryptionConfiguration) bool {
+	return ec.Resources[0].Providers[0].AESCBC != nil
+}
+
+// SetActive sets the EncryptionConfiguration to active or non-active (passive) state.
+// State active means that provider aescbc is the first in the list of providers.
+// State non-active (passive) means that provider identity is the first in the list of providers.
+func SetActive(ec *apiserverconfigv1.EncryptionConfiguration, active bool) error {
+	if IsActive(ec) == active {
+		// nothing to do
+		return nil
+	}
+	ec.Resources[0].Providers[0], ec.Resources[0].Providers[1] = ec.Resources[0].Providers[1], ec.Resources[0].Providers[0]
+	return nil
+}
+
+// CreatePassiveRotationKey adds a new key to the EncryptionConfiguration as a second key
+// in the list of keys. Note that the order matters and the first key in the list
+// of keys is used for encrypting etcd contents.
+func CreatePassiveRotationKey(ec *apiserverconfigv1.EncryptionConfiguration) error {
```

Comment:
Why create these functions if they are not implemented yet? This would make more sense if this was part of an interface but I'm also not sure whether the return values will stay like that.

Reply:
+1, let's remove them for now.